apps/Wallet/src/app/api/safe/util/utils.ts

```diff
@@ -3,6 +3,19 @@ import { ethers } from "ethers";
 import SafeApiKit from "@safe-global/api-kit";
 import Safe, { EthersAdapter } from "@safe-global/protocol-kit";
 import { SafeTransactionBody } from "@/types";
+import { etherscanId, network } from "./constants";
+
+export function getSigner() {
+  const provider = new ethers.providers.JsonRpcProvider(process.env.RPC_URL!);
```

Comment:
How do these functions differ from `getEthAdapter`?

Reply:
i want the signer, not the ethadapter (different objects are returned)
etherscansigner is added because using default signer was giving back cannot estimate gas errors, etherscansigner was not

Reply:
You can simply get the signer from the adapter by calling `.getSigner()`. Regarding etherscan should we not then just set it as the default instead of having two functions with different signers? I'd suggest keeping only the `getEthAdapter` adding the etherscan there and only using that to get the required setup for blockchain calls, aapart from that we should probably set the etherscanId in the .env

Reply:
What I find convoluted is having 3 functions that can all do exactly the same thing, there is no difference in the signer that comes from the adapter or the signer that comes from calling getSigner, the way I see it, as a developer, having one object that provides you with the necessities to interact with the blockchain is better than having 3 functions that can all do it.

Reply:
I'll leave it up to you

Reply:
ok, in utils theres a getEtherscanProvider and getSigner function now, which im exporting in the routes when we need signers and providers
i think this is the best way to do it